### DISCUSSION TITLE
U8x8 rotate text by 90 degrees

### DISCUSSION THREAD

**Discussion by epeschier** [olikraus/u8g2#2269] [Q&A] [answered]
I am using U8x8 lib on a 64x128 display. The text comes out potrait (so 8 characters per line, 16 lines).
Is there a way to rotate by 90 degrees? So the text is displayed in landscape mode?
Oliver confirmed that there is no method currently but perhaps anyone has found a way to do this?

**Comment by olikraus** [comment]
I actually have an idea for a workaround. I will try this on the weekend.

**Reply by olikraus** [reply]
Created an issue for this topic: https://github.com/olikraus/u8g2/issues/2271

**Reply by olikraus** [reply, marked as answer]
I have created a demo on how to rotate strings by 90 in u8x8:
https://github.com/olikraus/u8g2/blob/f50cb994708eb1e63c2efcc68685a08f7fcc988f/sys/arduino/u8x8/Rotate90/Rotate90.ino#L314-L349

The latest u8g2 beta is required for this:

You can download the latest U8g2 beta release from here: https://github.com/olikraus/U8g2_Arduino/archive/master.zip
Arduino IDE:
1. Remove the existing U8g2_Arduino library (https://stackoverflow.com/questions/16752806/how-do-i-remove-a-library-from-the-arduino-environment)
2. Install the U8g2_Arduino Zip file via Arduino IDE, add zip library menu (https://www.arduino.cc/en/Guide/Libraries).

PlatformIO:
`platformio.ini` (https://docs.platformio.org/en/latest/projectconf/section_env_library.html#lib-deps) should include 
```
lib_deps =
  u8g2=https://github.com/olikraus/U8g2_Arduino/archive/master.zip
```

**Reply by olikraus** [reply]
The example is also included in the latest beta.

**Reply by epeschier** [reply]
Works like a charm!
You only have to take into account that x and y are the same coordinates as when not rotating 90 degrees but you can do something like

`u8x8_draw_string_90(u8x8->getU8x8(),  MAX_LINES -y, x, text);`

Thanks Olivier!
